calicoctl/commands/resources.go

```diff
@@ -182,12 +183,31 @@ func executeConfigCommand(args map[string]interface{}, action action) commandRes
 
 	// Now execute the command on each resource in order, exiting as soon as we hit an
 	// error.
+	export := argutils.ArgBoolOrFalse(args, "--export")
+	nameSpecified := argutils.ArgStringOrBlank(args, "<NAME>")
 	for _, r := range resources {
 		res, err := executeResourceAction(args, client, r, action)
 		if err != nil {
 			results.err = err
 			break
 		}
+
+		for i, _ := range res {
+			// Remove the cluster specific metadata if the "--export" flag is specified
+			// Skip removing cluster specific metadata if this is is called as a "list"
+			// operation (no specific name is specified).
+			if export && nameSpecified != "" {
```

Comment:
this should still go before line 195 since these things won't change during the execution of this code (you will either have the flag and name or not for the calicoctl command, it's not going to change for each resources)